app/views/projects/_form.html.haml

```diff
@@ -15,24 +15,48 @@
 
   = f.fields_for :configs do |cf|
     - name,index  = cf.object.metric_name, cf.index
-    %fieldset{:id => name}
-      %legend= name.titleize
-      = hidden_field_tag "project[configs_attributes][#{index}][metric_name]", name
-      - cf.object.options.each_pair do |key,val|
-        - field_name = "project[configs_attributes][#{index}][options][#{key}]"
-        %label{:for => field_name}= "#{key}:"
-        = text_field_tag field_name, val, :disabled => @readonly
-        %br
-      - unless @readonly
-        = link_to 'Add new', '', :class => 'add'
-        .field.new
-          = text_field_tag "project[configs_attributes][#{index}][new][]", '', :id => nil, :class => 'newf'
-          \ :
-          = text_field_tag "project[configs_attributes][#{index}][new][]", '', :id => nil, :class => 'newf'
+    - classes = {'github' => ProjectMetricGithub, 'code_climate' => ProjectMetricCodeClimate, 'slack' => ProjectMetricSlack,'pivotal_tracker' => ProjectMetricPivotalTracker,'slack_trends' => ProjectMetricSlackTrends}
+    - if classes[name].respond_to?(:credentials)
+      %fieldset{:id => name}
+        %legend= name.titleize
+        = hidden_field_tag "project[configs_attributes][#{index}][metric_name]", name
+        - existed = []
```

Comment:
edit and index? you mean edit and new right? even if they are you can still pull all this into instance variables that you set in the appropriate controller actions and then minimize the code logic in the view itself